CMakeLists.txt

```diff
@@ -287,6 +287,10 @@ option(WITH_ASYNC_EXPORT_PREVIEW "Whether to enable async export" OFF)
 option(WITH_METRICS_EXEMPLAR_PREVIEW
        "Whether to enable exemplar within metrics" OFF)
 
+option(OPENTELEMETRY_STATIC_ONLY
```

Comment:
should we name it to (say) - `OPENTELEMETRY_SKIP_DYNAMIC_LOADING_TESTS` - if this is to skip generating libraries for specific tests. Or any other name, OPENTELEMETRY_STATIC_ONLY is too generic.